docs/usingtheapi.md

```diff
@@ -1,12 +1,65 @@
 # Using the API
 
-These examples assume that the REMS instance you want to talk to is running locally at `localhost:3000`.
+## Documentation
+
+The REMS API is documented using
+[OpenAPI aka Swagger](https://swagger.io/docs/specification/about/).
+You can check out the API docs using
+[the swagger-ui of the public demo instance](https://rems-demo.rahtiapp.fi/swagger-ui),
+or your local development instance at <http://localhost:3000/swagger-ui>.
 
 ## Authentication
 
-To call the API programmatically, you will first need to add an API key to the `api_key` database table. The API key must be provided in the `x-rems-api-key` header when calling the API.
+You can use either browser session authentication or an API key.
+
+### Session authentication
+
+Once a user has logged in to REMS normally, they can use the API from
+their browser. In practice this means that you can log in and then use
+the "Try it out!" feature in Swagger UI.
+
+Session authentication is also used by implementation of  REMS browser UI
+
+### API key authentication
+
+You can also authenticate by setting these two headers:
+- `x-rems-api-key` -- an API key
+- `x-rems-user-id` -- the user id of a user to impersonate
+
+API keys can be defined and modified using the `api-key` command line option to REMS. Here are some examples:
+
+```sh
+# Get all API keys:
+java -Drems.config=path/to/config -jar rems.jar api-key get
+# Add an API key with an optional comment:
+java -Drems.config=path/to/config -jar rems.jar api-key add abcd1234 this is my secret api key
+```
 
```

Comment:
Do we also have api-key remove?

Reply:
no, I didn't implement it. good idea though